### DISCUSSION TITLE
Order of nodes in data or heterodata from DataLoader

### DISCUSSION THREAD

**Discussion by Nokimann** [pyg-team/pytorch_geometric#3939] [Q&A] [answered]
I wonder if the nodes' orders keep the same in a mini-batch of data or heterodata from DataLoader.

For example,
Graph1 has three nodes with features 1-1, 1-2, and 1-3, and 
Graph2 has two nodes with features 2-1 and 2-2.

The mini-batch contains five nodes from Graph1 and Graph2.
Are the nodes' orders of Graph1 and Graph2 in the mini-batch still the same? (not 1-3, 1-1, and 1-2 and/or not 2-2, and 2-1)
Thus, the  nodes' orders of nodes in the mini-batch are 1-1, 1-2, 1-3, 2-1, and 2-2 (or 2-1, 2-2, 1-1, 1-2, and 1-3 if shuffled)

**Comment by rusty1s** [comment, marked as answer]
Yes, exactly. The nodes' order is kept the same, that is, node feature matrices are simply concatenated in the node dimension.

**Reply by Nokimann** [reply]
Because of  `__cat_dim__` in `Data`, orders of edge_index and edge_attr are also kept!
